src/StreamJsonRpc/DataContracts/JsonRpcError.cs

```diff
@@ -14,16 +15,16 @@ internal JsonRpcError(int code, string message)
         }
 
         [JsonConstructor]
-        internal JsonRpcError(int code, string message, dynamic data)
+        internal JsonRpcError(int code, string message, JObject data)
         {
             this.Code = code;
             this.Message = message;
             this.Data = data;
         }
 
-        public string ErrorStack => this.Data?.stack;
+        public string ErrorStack => this.Data?["stack"]?.Type == JTokenType.String ? this.Data.Value<string>("stack") : null;
```

Comment:
These names repeat in several places. It would be better to use string constants for them.